vm/compiler/src/program/finalize/command/load.rs

```diff
@@ -0,0 +1,222 @@
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// The snarkVM library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with the snarkVM library. If not, see <https://www.gnu.org/licenses/>.
+
+use crate::{FinalizeRegisters, Opcode, Operand, ProgramStorage, ProgramStore, Stack};
+use console::{
+    network::prelude::*,
+    program::{Identifier, Register, Value},
+};
+
+// TODO: Verify that the desired format is `load accounts[r0] into r1;`.
+
+/// Loads the value stored at `operand` in `mapping` into `destination`.
+#[derive(Clone, PartialEq, Eq, Hash)]
+pub struct Load<N: Network> {
+    /// The mapping name.
+    mapping: Identifier<N>,
+    /// The key to access the mapping.
+    key: Operand<N>,
+    /// The destination register.
+    destination: Register<N>,
+}
+
+impl<N: Network> Load<N> {
+    /// Returns the opcode.
+    #[inline]
+    pub const fn opcode() -> Opcode {
+        Opcode::Command("load")
+    }
+
+    /// Returns the operands in the operation.
+    #[inline]
+    pub fn operands(&self) -> Vec<Operand<N>> {
+        vec![self.key.clone()]
+    }
+
+    /// Returns the mapping name.
+    #[inline]
+    pub const fn mapping_name(&self) -> &Identifier<N> {
+        &self.mapping
+    }
+
+    /// Returns the operand containing the key.
+    #[inline]
+    pub const fn key(&self) -> &Operand<N> {
+        &self.key
+    }
+
+    /// Returns the destination register.
+    #[inline]
+    pub const fn destination(&self) -> &Register<N> {
+        &self.destination
+    }
+}
+
+impl<N: Network> Load<N> {
+    /// Evaluates the command.
+    #[inline]
+    pub fn evaluate_finalize<P: ProgramStorage<N>>(
+        &self,
+        stack: &Stack<N>,
+        store: &ProgramStore<N, P>,
+        registers: &mut FinalizeRegisters<N>,
+    ) -> Result<()> {
+        // Ensure the mapping exists in storage.
+        if !store.contains_mapping(stack.program_id(), &self.mapping)? {
+            bail!("Mapping '{}/{}' does not exist in storage", stack.program_id(), self.mapping);
+        }
+
+        // Load the operand as a plaintext.
+        let key = registers.load_plaintext(stack, &self.key)?;
+
+        // Retrieve the value from storage as a literal.
+        let value = match store.get_value(stack.program_id(), &self.mapping, &key)? {
+            Some(Value::Plaintext(plaintext)) => Value::Plaintext(plaintext),
+            Some(Value::Record(..)) => bail!("Cannot 'load' a 'record'"),
+            // If the key does not exist, use a default value.
+            // TODO: Infer type.
+            None => todo!(),
```

Comment:
When attempting to load a value from an uninitialized `key`, we should be bailing as the entry in the `mapping` does not exist.